internal/resources/resource_host.go

```diff
@@ -587,9 +597,16 @@ func resourceMetalHostRead(d *schema.ResourceData, meta interface{}) (err error)
 		return err
 	}
 
-	d.Set(hCHAPUser, host.ISCSIConfig.CHAPUser)
-	d.Set(hCHAPSecret, host.ISCSIConfig.CHAPSecret)
-	d.Set(hInitiatorName, host.ISCSIConfig.InitiatorName)
+	if protocol == "iscsi" {
+		_ = d.Set(hCHAPUser, host.ISCSIConfig.CHAPUser)
+		_ = d.Set(hCHAPSecret, host.ISCSIConfig.CHAPSecret)
+		_ = d.Set(hInitiatorName, host.ISCSIConfig.InitiatorName)
+	}
+
+	if protocol == "fc" {
+		_ = d.Set(hWWPNS, host.WWPNs)
+	}
+
```

Comment:
why set wwpn is related to protocol? regardless protocol, wwpn should be set.

Reply:
Will revert it.